Control/lib/services/Lock.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+*/
+const {WebSocketMessage, Log} = require('@aliceo2/web-ui');
+const log = new Log(`${process.env.npm_config_log_label ?? 'cog'}/lockservice`);
+
+/**
+ * Model representing the lock of the UI, one owner at a time
+ */
+class Lock {
+  /**
+   * Initialize lock as free / unlocked.
+   */
+  constructor() {
+    this.lockedBy = {};
+    this.lockedByName = {};
+  }
+
+  /**
+   *  Sets WebSocket instance
+   *  @param {object} ws
+   */
+  setWs(ws) {
+    this.webSocket = ws;
+  }
+
+  /**
+   * Provides state of all locks
+   * @returns {object}
+   */
+  state() {
+    return {
+      lockedBy: this.lockedBy,
+      lockedByName: this.lockedByName
+    }
+  }
+
+  /** 
+   * Send to all users state of Pad via Websocket
+   */
+  broadcastLockState() {
+    this.webSocket.broadcast(new WebSocketMessage().setCommand('padlock-update').setPayload(this.state()));
+  } 
+
+  /** 
+   * Method to try to acquire lock
+   * @param {Request} req
+   * @param {Response} res
+   */
+  lockDetector(req, res) {
+    try {
+      if (!('name' in req.body)) {
+        throw new Error('Unspecified lock entity');
+      }
+      const entity = req.body.name;
```

Comment:
You can directly check for `name`
```suggestion
      const entity = req.body?.name;
      if (!entity) {
        throw new Error('Unspecified lock entity');
      } 
```